### PR TITLE
[core] speed up tagged snapshot search

### PR DESCRIPTION
**Pull request by askwang**
### Purpose

Linked issue: close #xxx

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/utils/TagManager.java [hunk 1]
```diff
@@ -355,9 +355,16 @@ public List<String> allTagNames() {
...
+        int left = 0;
+        int right = taggedSnapshots.size() - 1;
+        while (left <= right) {
+            int mid = left + (right - left) / 2;
+            if (taggedSnapshots.get(mid).id() == taggedSnapshot.id()) {
+                return mid;
+            } else if (taggedSnapshots.get(mid).id() < taggedSnapshot.id()) {
+                left = mid + 1;
+            } else {
+                right = mid - 1;
```
Comment (TaoZex) [inline paimon-core/src/main/java/org/apache/paimon/utils/TagManager.java:367]: How about encapsulating the binary search into a general static method?

Comment (askwang) [reply paimon-core/src/main/java/org/apache/paimon/utils/TagManager.java:367]: good idea. should we add a new util class `ListUtils` or do you have some advice?

**Review by JingsongLi — CHANGES_REQUESTED** [review]
Hi @askwang , I think we don't need to optimize search here. Here is just loop in memory. Very very fast...

**Comment by askwang** [comment]
> Hi @askwang , I think we don't need to optimize search here. Here is just loop in memory. Very very fast...

+1, I close it

**Pull request closed without merging**